internal/pkg/service/common/utctime/utctime.go

```diff
@@ -61,7 +61,7 @@ func FormatTime(t time.Time) string {
 }
 
 func From(t time.Time) UTCTime {
-	return UTCTime(t.UTC().Truncate(time.Millisecond))
+	return UTCTime(t.UTC())
 }
```

Comment:
Small unrelated fix:

This broke `UTCTime.Time().Equal(now)` comparison.

On marshalling, microseconds (all parts smaller than millisecond) are truncated by the specified format.

The format ends with 3 `.000` millisecond digits.

Reply:
Isnt the problem of comparison side instead of function side? You can truncate `time.Now`. I would expect that the test uses truncate as well.